Helper/Marketplace/WebkulHelper.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+/**
+ * Class AbstractHelper
+ *
+ * @author      Open Source Team
+ * @copyright   2021 Pagar.me (https://pagar.me)
+ * @license     https://pagar.me Copyright
+ *
+ * @link        https://pagar.me
+ */
+
+namespace Pagarme\Pagarme\Helper\Marketplace;
+
+use Magento\Framework\App\ObjectManager;
+use Pagarme\Core\Kernel\Services\MoneyService;
+use Pagarme\Core\Marketplace\Services\RecipientService;
+use Webkul\Marketplace\Helper\Payment;
+use Webkul\Marketplace\Helper\Data;
+
+class WebkulHelper
+{
+    private $productCollectionFactory;
+    private $salesPerPartnerCollectionFactory;
+    private $objectManager;
+    private $recipientService;
+
+    private $enabled = false;
+
+    public function __construct() {
+
+        $this->objectManager = ObjectManager::getInstance();
+
+        if (!class_exists('Webkul\\Marketplace\\Helper\\Payment')) {
+            return null;
+        }
+
+        $this->mpPaymentHelper = $this->objectManager->get(Payment::class);
+
+        $this->setEnabled(true);
+
+        $this->recipientService = new RecipientService();
+        $this->moneyService = new MoneyService();
+    }
+
+    public function isEnabled()
+    {
+        return $this->enabled;
+    }
+
+    public function setEnabled($enabled)
+    {
+        $this->enabled = $enabled;
+    }
+
+    public function getSplitDataFromOrder($platformOrder)
+    {
+        $orderItems = $platformOrder->getAllItems();
+        $splitData['sellers'] = [];
+        $totalPaid = (int) $this->moneyService->floatToCents(
+            $platformOrder->getGrandTotal()
+        );
+
+        foreach ($orderItems as $item) {
+
+            $productId = $item->getProductId();
+            $sellerDetail = $this->mpPaymentHelper->getSellerDetail($productId);
+            $sellerId = $sellerDetail['id'];
+            $itemPrice = (int) $this->moneyService->floatToCents(
+                $item->getRowTotal()
+            );
+
+            if(!$sellerId) {
+                continue;
+            }
+
+            $splitData = $this->sumSellersCommissionsIfArrayKeyExists(
```

Comment:
Eu entendi como estão sendo usadas as funções `sumSellersCommissionsIfArrayKeyExists` e `setNewKeyIfArrayKeyNotExists`, mas acho que a gente consegue matar elas fazendo um fluxo tipo esse: 
```suggestion
            $sellerId = $sellerDetail['id'];
            $sellerDataForProduct = $this->getSellerData(
                $itemPrice,
                $sellerDetail
            );
            if (array_key_exists($sellerId, $splitData['sellers'])) {
               $splitData['sellers'][$sellerId]['sellerCommission'] += $sellerDataForProduct['sellers'][$sellerId]['sellerCommission'];
               continue;
            }
             $splitData['sellers'][$sellerId] = $sellerDataForProduct;
```

Reply:
corrigido!